core/src/main/scala/org/apache/spark/util/ThreadUtils.scala

```diff
@@ -157,23 +259,23 @@ private[spark] object ThreadUtils {
    */
   def newDaemonFixedThreadPool(nThreads: Int, prefix: String): ThreadPoolExecutor = {
     val threadFactory = namedThreadFactory(prefix)
-    Executors.newFixedThreadPool(nThreads, threadFactory).asInstanceOf[ThreadPoolExecutor]
+    MDCAwareThreadPoolExecutor.newFixedThreadPool(nThreads, threadFactory)
```

Comment:
After a second thought, I'm wondering what's the use case for this change.

For example, `newDaemonFixedThreadPool` is used in `TaskResultGetter`. I don't think we can get MDC properties there. In fact, we only set the MDC properties in `Executor.run`, and these thread pools are mostly created by the driver.

Reply:
1. What do you mean can get MDS properties? we can always get them do you mean set? for now it is not used but later it great you can set trace data in MDC and get it in logs.
2. It will be great to add MDC data for the driver also. I will create new PR for that after that one will be merged.

Reply:
I mean the things we documented: 1) task name apparently doesn't apply to the driver side 2) custom MDC properties set by `spark.sparkContext.setLocalProperty("mdc." + name, "value")`

We usually don't commit changes that "may be used later". We can submit the changes when we really need them.

Reply:
Does the only usage of `newDaemonFixedThreadPool` is on the driver side?

Reply:
also, we can add to the docs the MDC is propagated into all threads on the driver and that also very useful

Reply:
I only find one executor use case: `ContinuousCoalesceRDD`, which is for continuous streaming. But the runnable submitted to the thread pool does not have logging.

I checked some other methods in `ThreadUtils` but couldn't find executor usages that need logging.

What's your target use case? Usually a Spark task means one thread, so it's very rare to use a thread pool within a Spark task,

Reply:
in our yarn env, we find that without all the changes in the ThreadUtils not all the logs in the executor get the right MDC only after I add all we get the MDC data in all logs.

Reply:
Do you use open source Spark or your own fork?

It's hard to convince the reviewers that the change is necessary, without clearly pointing out where we create these thread pools at executor side. Can you investigate further?

Reply:
I will check all usages.
for now, we are using our own fork as we can't work without that fix.

Reply:
I check and all uses of that function are from driver code. 
What do you think about reverting that one and create a new one called `newMDCAwareDaemonFixedThreadPool` so it can be used later?

Reply:
We don't commit dead code that no place uses. Please only leave the code that is effective.

Reply:
OK